tests/store/test_organism_filters.py

```diff
@@ -0,0 +1,48 @@
+from cg.store.api.core import Store
+from cg.store.models import Organism
+from cg.store.organism_filters import filter_organism_by_internal_id
+
+
+def test_filter_organism_by_internal_id_returns_correct_organism(store_with_organisms: Store):
+    """Test filtering an organism by internal ID when the ID exists."""
+
+    # GIVEN a store with multiple organisms
+    num_organisms = store_with_organisms._get_organism_query().count()
+    assert num_organisms > 0
+
+    # Select a random organism from the store
+    organism = store_with_organisms.query(Organism).first()
```

Comment:
Maybe?
```suggestion
    # GIVEN a store with multiple organisms
    organisms: Query = store_with_organisms._get_organism_query()
    num_organisms = organisms.count()
    assert num_organisms > 0

    # GIVEN a random organism from the store
    organism = organisms.first()
```

Reply:
another option would be use a `List[Organism]` instead of `Query` and `len()` insterad of ´.count()`.